src/dune_rules/pkg_toolchain.ml

```diff
@@ -27,13 +27,13 @@ let pkg_dir (pkg : Dune_pkg.Lock_dir.Pkg.t) =
      way). *)
   let dir_name =
     (* TODO should include resolved deps *)
-    let pkg_hash = Digest.generic (Lock_dir.Pkg.remove_locs pkg) in
+    let pkg_hash = Lock_dir.Pkg.hash (Lock_dir.Pkg.remove_locs pkg) in
```

Comment:
Can't we keep using a cryptographic hash function? Regular hash functions like this can easily collide.

Reply:
Done. I've extended the `Digest` module a little to make it easier to thread a hash function through custom types.